tools/acorn-optimizer.mjs

```diff
@@ -1288,28 +1288,27 @@ function littleEndianHeap(ast) {
   });
 }
 
-// Instrument heap accesses to call GROWABLE_HEAP_* helper functions instead, which allows
+// Instrument heap accesses to call growMemViews helper function, which allows
 // pthreads + memory growth to work (we check if the memory was grown on another thread
 // in each access), see #8365.
 function growableHeap(ast) {
   recursiveWalk(ast, {
     FunctionDeclaration(node, c) {
-      // Do not recurse into to GROWABLE_HEAP_ helper functions themselves.
+      // Do not recurse into the helper function itself.
       if (
         !(
           node.id.type === 'Identifier' &&
-          (node.id.name.startsWith('GROWABLE_HEAP_') || node.id.name === 'LE_HEAP_UPDATE')
+          (node.id.name === 'growMemViews' || node.id.name === 'LE_HEAP_UPDATE')
         )
       ) {
         c(node.body);
       }
     },
     AssignmentExpression: (node) => {
-      if (node.left.type === 'Identifier' && isEmscriptenHEAP(node.left.name)) {
+      if (node.left.type !== 'Identifier') {
         // Don't transform initial setup of the arrays.
```

Comment:
This comment looks stale.

Reply:
It's still relevant - that's the main usecase we want to cover here by skipping the lhs - but I can clarify the comment.